src/main/kotlin/org/jetbrains/research/libsl/nodes/references/TypeReference.kt

```diff
@@ -70,81 +167,64 @@ abstract class TypeReference(
 
         return true
     }
+}
 
-    override fun isSameReference(other: TypeReference): Boolean {
-        return this.name == other.name
-                && this.isPointer == other.isPointer
-                && (other.genericReferences.isEmpty() || this.genericReferences.isSameReference(other.genericReferences))
+
+data class PlainTypeReference(
+    val name: String,
+    val isPointer: Boolean,
+    var typeBound: GenericTypeBound = GenericTypeBound.EMPTY,
```

Comment:
If this is a reference to a *plain* (read *simple*) type, then why is there a `typeBound` field?

Reply:
Fixed;
This field was deleted;